common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -168,7 +168,8 @@ public static Level getLevel() {
             Collections.newSetFromMap(new ConcurrentHashMap<DefaultResourceLeak<?>, Boolean>());
 
     private final ReferenceQueue<Object> refQueue = new ReferenceQueue<Object>();
-    private final ConcurrentMap<String, Boolean> reportedLeaks = PlatformDependent.newConcurrentHashMap();
+    private final Set<String> reportedLeaks =
+            Collections.newSetFromMap(new ConcurrentHashMap<String, Boolean>());
 
```

Comment:
One could also argue that newSetFromMap has additional overhead, memory and stack (all methods are delegated to the backing map of the "set").

Reply:
Hi, johnou
This does not seem to be a serious problem. Because `reportedLeaks` only takes effect when a leak occurs, the application will not be in a leaked state for a long time. This change seems to only result in the addition of a Set object to the stack.